starter_code/src/movies.js

```diff
@@ -1,20 +1,131 @@
 /* eslint no-restricted-globals: 'off' */
 // Turn duration of the movies from hours to minutes 
+function minutesCalc(str) {
+    var newArr = str.split(" ");
+    var hours = Number((newArr[0].split("h"))[0]);
+    if (newArr.length > 1) {
+        var duration = hours * 60 + Number((newArr[1].split("min"))[0]);
+    } else {
+        var duration = hours * 60;
+    }
+    return duration;
+}
 
+function turnHoursToMinutes(arr) {
+    var converMovies = arr.map(function (a) {
+        var b={};
+        for(key in a){
+            b[key] = a[key]                      
+        }
+        b.duration = minutesCalc(a.duration);          
+        return b;
+    })
+    return converMovies;
+}
 
 // Get the average of all rates with 2 decimals 
-
+function ratesAverage(arr) {
+    var total = arr.reduce(function (accumulator, item) {
+        return accumulator + Number(item.rate);
+    }, 0);
+    return parseFloat((total / arr.length).toFixed(2));
+};
 
 // Get the average of Drama Movies
-
+function dramaMoviesRate(arr) {
+    var dramaMovies = arr.filter(function (item) {
+        return item.genre.find(function (a) {return a === "Drama";}) 
+        === "Drama";
+    });
+    if (dramaMovies.length > 0) {
+        return ratesAverage(dramaMovies);
+    }
+}
 
 // Order by time duration, in growing order
-
+function orderByDuration(arr) {
+    //var newArr = turnHoursToMinutes(arr); <--- What?
+    return arr.sort(function(a,b) {
+        if (a.duration > b.duration) {
+            return 1;
+        }
+        else if (b.duration > a.duration) {
+            return -1;
+        } 
+        else {
+            if (a.title > b.title) {
+                return 1;
+            } else if (b.title > a.title) {
+                return -1;
+            } else {
+                return 0;
+            }
+        }
+    });
+}
 
 // How many movies did STEVEN SPIELBERG
 
+function howManyMovies(arr) {
+    var directedMovies = arr.filter(function(element) {
+        return (element.director === 'Steven Spielberg' && element.genre.indexOf('Drama') > -1);
+    })
+    if (arr.length > 0) {
+        return 'Steven Spielberg directed ' + directedMovies.length + ' drama movies!';
+    }
+    
+}
 
 // Order by title and print the first 20 titles
-
+function orderAlphabetically(arr) {
+    arr.sort(function(a,b) {
+        if(a.title > b.title) {
+            return 1;
+        } 
+        else if(b.title > a.title) {
+            return -1;
+        } else {
+            return 0;
+        }
+    });
+    return arr.reduce(function(accumulator, item, index) {
```

Comment:
This is a clever solution, but keep in mind something: if you have one million of items, the code will run one million times, when you only need 20 iterations.